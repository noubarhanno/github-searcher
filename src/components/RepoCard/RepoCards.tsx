import React, { FC } from 'react';
import { Flex } from 'reflexbox';
import RepoCard from './RepoCard';
// import { CardWrapper } from './RepoCard.styled';
import { IRepo } from 'pages/Repos/redux/repos.interface';

export type TProps = {
    userData: IRepo[];
};
const UserCards: FC<TProps> = ({ userData }) => {
    return (
        <Flex flexDirection="row" alignItems="center" width="100%" flexWrap="wrap">
            {userData.map((data: IRepo, index: number) => (
                <RepoCard userData={data} key={index} />
            ))}
        </Flex>
    );
};

export default UserCards;
