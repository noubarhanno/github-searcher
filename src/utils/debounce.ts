let timeout: NodeJS.Timeout | null;
export default (func: any, wait: number, immediate: boolean = false) => {
    return function (...args: any) {
        // @ts-ignore
        const context: any = this;
        const later = () => {
            timeout = null;
            if (!immediate) {
                func.apply(context, args);
            }
        };
        const callNow = immediate && !timeout;
        // @ts-ignore
        clearTimeout(timeout);
        // @ts-ignore
        timeout = setTimeout(later, wait);
        if (callNow) {
            func.apply(context, args);
        }
    };
};
