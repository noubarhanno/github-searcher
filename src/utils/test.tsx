import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import ThemeProvider from 'components/themeProvider';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import initialAppState from 'store/initialState';
import thunk from 'redux-thunk';

const AllTheProviders: React.FC = (props) => <ThemeProvider includeReset={false} {...props} />;

const customRender = (ui: React.ReactElement, options?: RenderOptions) => {
    return render(ui, { wrapper: AllTheProviders, ...options });
};

const renderWithRedux = (component: React.ReactNode, storeState = {}): any => {
    const getStore = createMockStore([thunk]);
    const store = getStore({ ...initialAppState, ...storeState });
    return {
        ...customRender(
            <Provider store={store}>
                <ThemeProvider includeReset={false}>{component}</ThemeProvider>
            </Provider>,
        ),
        store,
    };
};

/**
 * This method creates a mock store for testing purposes
 * @param state : any redux state
 */
export const getMockStore = (state?: any) => {
    const mockStore = createMockStore([thunk]);
    return mockStore({ ...initialAppState, ...state });
};

export * from '@testing-library/react';
export { customRender as render, renderWithRedux };
