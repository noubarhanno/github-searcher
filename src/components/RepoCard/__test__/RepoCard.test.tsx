import React from 'react';
import { render } from 'utils/test';
import RepoCards from '../RepoCards';
import mockStore from './__mocks__';
import ThemeProvider from '../../themeProvider';

describe('repo card test suite', () => {
    it('should match the snapshot', () => {
        const wrapper = render(
            <ThemeProvider>
                <RepoCards userData={mockStore.userData} />
            </ThemeProvider>,
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('should show 3 cards', () => {
        const { getAllByTestId } = render(
            <ThemeProvider>
                <RepoCards userData={mockStore.userData} />
            </ThemeProvider>,
        );
        const cards = getAllByTestId('repo-card');
        expect(cards.length).toEqual(3);
    });
    it('should show 2 cards with badge and one without ', () => {
        const { getAllByTestId } = render(
            <ThemeProvider>
                <RepoCards userData={mockStore.userData} />
            </ThemeProvider>,
        );
        const cards = getAllByTestId('language-badge');
        expect(cards.length).toEqual(2);
    });
});
