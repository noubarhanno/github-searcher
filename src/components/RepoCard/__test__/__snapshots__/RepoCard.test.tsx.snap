// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repo card test suite should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1xexys8"
      >
        <div
          class="css-1k5ovvj"
          data-testid="repo-card"
          elevation="2"
        >
          <div
            class="css-l3rx45"
          >
            <div
              class="css-19mood2"
            >
              <a
                class="css-1r0izxd"
                href="/"
                target="_blank"
              >
                full name
              </a>
            </div>
            <div
              class="css-1cc7kk0"
            >
              <div>
                <p
                  class="css-8rkuw"
                >
                  Description
                </p>
                <div
                  class="css-1nemtvz"
                >
                  awesome repo
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-8m3gtl"
          >
            <div>
              <p
                class="css-8rkuw"
              >
                Stars: 
                <span
                  class="css-mhms3j"
                >
                   10
                </span>
              </p>
            </div>
            <div>
              <a
                class="css-1r0izxd"
                href="/"
              >
                Author: 
                <span
                  class="css-mhms3j"
                >
                   noubar
                </span>
              </a>
            </div>
          </div>
          <div
            class="css-1fj1bi0"
          >
            <div
              class="css-p6s3br"
              data-testid="language-badge"
            >
              javascript
            </div>
          </div>
        </div>
        <div
          class="css-1k5ovvj"
          data-testid="repo-card"
          elevation="2"
        >
          <div
            class="css-l3rx45"
          >
            <div
              class="css-19mood2"
            >
              <a
                class="css-1r0izxd"
                href="/"
                target="_blank"
              >
                full name
              </a>
            </div>
            <div
              class="css-1cc7kk0"
            >
              <div>
                <p
                  class="css-8rkuw"
                >
                  Description
                </p>
                <div
                  class="css-1nemtvz"
                >
                  awesome repo
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-8m3gtl"
          >
            <div>
              <p
                class="css-8rkuw"
              >
                Stars: 
                <span
                  class="css-mhms3j"
                >
                   10
                </span>
              </p>
            </div>
            <div>
              <a
                class="css-1r0izxd"
                href="/"
              >
                Author: 
                <span
                  class="css-mhms3j"
                >
                   noubar
                </span>
              </a>
            </div>
          </div>
          <div
            class="css-1fj1bi0"
          >
            <div
              class="css-p6s3br"
              data-testid="language-badge"
            >
              javascript
            </div>
          </div>
        </div>
        <div
          class="css-1k5ovvj"
          data-testid="repo-card"
          elevation="2"
        >
          <div
            class="css-l3rx45"
          >
            <div
              class="css-19mood2"
            >
              <a
                class="css-1r0izxd"
                href="/"
                target="_blank"
              >
                full name
              </a>
            </div>
            <div
              class="css-1cc7kk0"
            >
              <div>
                <p
                  class="css-8rkuw"
                >
                  Description
                </p>
                <div
                  class="css-1nemtvz"
                >
                  awesome repo
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-8m3gtl"
          >
            <div>
              <p
                class="css-8rkuw"
              >
                Stars: 
                <span
                  class="css-mhms3j"
                >
                   10
                </span>
              </p>
            </div>
            <div>
              <a
                class="css-1r0izxd"
                href="/"
              >
                Author: 
                <span
                  class="css-mhms3j"
                >
                   noubar
                </span>
              </a>
            </div>
          </div>
          <div
            class="css-1fj1bi0"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1xexys8"
    >
      <div
        class="css-1k5ovvj"
        data-testid="repo-card"
        elevation="2"
      >
        <div
          class="css-l3rx45"
        >
          <div
            class="css-19mood2"
          >
            <a
              class="css-1r0izxd"
              href="/"
              target="_blank"
            >
              full name
            </a>
          </div>
          <div
            class="css-1cc7kk0"
          >
            <div>
              <p
                class="css-8rkuw"
              >
                Description
              </p>
              <div
                class="css-1nemtvz"
              >
                awesome repo
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-8m3gtl"
        >
          <div>
            <p
              class="css-8rkuw"
            >
              Stars: 
              <span
                class="css-mhms3j"
              >
                 10
              </span>
            </p>
          </div>
          <div>
            <a
              class="css-1r0izxd"
              href="/"
            >
              Author: 
              <span
                class="css-mhms3j"
              >
                 noubar
              </span>
            </a>
          </div>
        </div>
        <div
          class="css-1fj1bi0"
        >
          <div
            class="css-p6s3br"
            data-testid="language-badge"
          >
            javascript
          </div>
        </div>
      </div>
      <div
        class="css-1k5ovvj"
        data-testid="repo-card"
        elevation="2"
      >
        <div
          class="css-l3rx45"
        >
          <div
            class="css-19mood2"
          >
            <a
              class="css-1r0izxd"
              href="/"
              target="_blank"
            >
              full name
            </a>
          </div>
          <div
            class="css-1cc7kk0"
          >
            <div>
              <p
                class="css-8rkuw"
              >
                Description
              </p>
              <div
                class="css-1nemtvz"
              >
                awesome repo
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-8m3gtl"
        >
          <div>
            <p
              class="css-8rkuw"
            >
              Stars: 
              <span
                class="css-mhms3j"
              >
                 10
              </span>
            </p>
          </div>
          <div>
            <a
              class="css-1r0izxd"
              href="/"
            >
              Author: 
              <span
                class="css-mhms3j"
              >
                 noubar
              </span>
            </a>
          </div>
        </div>
        <div
          class="css-1fj1bi0"
        >
          <div
            class="css-p6s3br"
            data-testid="language-badge"
          >
            javascript
          </div>
        </div>
      </div>
      <div
        class="css-1k5ovvj"
        data-testid="repo-card"
        elevation="2"
      >
        <div
          class="css-l3rx45"
        >
          <div
            class="css-19mood2"
          >
            <a
              class="css-1r0izxd"
              href="/"
              target="_blank"
            >
              full name
            </a>
          </div>
          <div
            class="css-1cc7kk0"
          >
            <div>
              <p
                class="css-8rkuw"
              >
                Description
              </p>
              <div
                class="css-1nemtvz"
              >
                awesome repo
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-8m3gtl"
        >
          <div>
            <p
              class="css-8rkuw"
            >
              Stars: 
              <span
                class="css-mhms3j"
              >
                 10
              </span>
            </p>
          </div>
          <div>
            <a
              class="css-1r0izxd"
              href="/"
            >
              Author: 
              <span
                class="css-mhms3j"
              >
                 noubar
              </span>
            </a>
          </div>
        </div>
        <div
          class="css-1fj1bi0"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
